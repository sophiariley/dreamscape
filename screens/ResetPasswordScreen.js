import React, { useState, useEffect} from "react";
import { ImageBackground, KeyboardAvoidingView, StyleSheet, Text, TextInput, TouchableOpacity, View } from "react-native";
import {useNavigation} from '@react-navigation/core'
import LoginScreen from "./LoginScreen";

const ResetPasswordScreen = ({navigation}) => {
    const [email, setEmail] = useState('');

    return (
        <View style={styles.container}>
            <Text style={styles.resetPassword}> Reset Password </Text>

            <View style={styles.inputContainer}>
                <Text style={styles.miscText}>Enter the email associated with your</Text>
                <Text style={styles.miscText}>account:</Text>
                <TextInput 
                    style={styles.inputText} 
                    placeholder="Email"
                    value={email}
                    onChangeText={text => setEmail(text)}
                />
            </View>

            <View style={styles.resetPasswordButtonContainer}>
                <TouchableOpacity
                    onPress={() => {{navigation.navigate('Login')}}}
                    style={styles.resetPasswordButton}
                > 
                <Text style={styles.resetPasswordButtonText}>Reset Password</Text>
                </TouchableOpacity>
            </View>

            <View>
                <TouchableOpacity
                    onPress={() => {{navigation.navigate('Login')}}}
                >
                    <Text style={styles.returnToLogin}>Return to login</Text>
                </TouchableOpacity>
            </View>

        </View>
    )
}

export default ResetPasswordScreen;

const styles = StyleSheet.create ({
    container: {
        backgroundColor: 'white',
        height: '100%',
        width: '100%',
        alignContent: 'center',
        justifyContent: 'center'
    },  
    inputContainer: {
        justifyContent: 'center',
        alignItems: 'center',
        top: 200
    },
    resetPassword: {
        color: '#F8C98A',
        fontSize: 40,
        alignSelf: 'center',
        top: 225
    },
    resetPasswordButtonContainer: {
        width: '70%',
        height: '20%',
        flex: 1,
        alignSelf: 'center',
        justifyContent: 'center'
    },
    resetPasswordButton: {
        backgroundColor: '#D28A8E', // pink color
        width: '100%',
        padding: 15,
        borderRadius: 5,
        alignItems: 'center',
    },
    resetPasswordButtonText: {
        fontSize: 20,
        color: '#F6F6F6'
    },
    miscText: {
        color: '#3A6496',
        alignSelf: 'flex-start',
        left: 40,
        fontSize: 20
    },
    inputText: {
        fontSize: 20,
        backgroundColor: '#DADADA',
        opacity: .5,
        height: '20%',
        width: '80%',
        borderRadius: 5,
        marginTop: 5,
        marginBottom: 10,
        paddingLeft: 10
    },
    returnToLogin: {
        alignSelf: 'center',
        color: '#3A6496',
        bottom: 100,
        fontSize: 20
    }
})