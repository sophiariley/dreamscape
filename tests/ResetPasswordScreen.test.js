import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react-native'; 
import ResetPasswordScreen from './ResetPasswordScreen';
import {create} from 'react-test-renderer';

// basic rendering test
test('renders correctly across screens', () => {
  const tree = create(<ResetPasswordScreen />);
  expect(tree).toMatchSnapshot();
});


// Testing email input box

const EMAIL_TEXT = 'email';

describe('Testing email input', () => {
  it('should find the email box via testId', () => {
  const testIdName = 'emailBox';
  const {getByTestId} = render(<ResetPasswordScreen />);
  const foundButton = getByTestId(testIdName);
  expect(foundButton).toBeTruthy();
  });

    it('Email box is pressable', () => {
      const page = render(<ResetPasswordScreen/>);
      const emailBox = page.getByTestId('emailBox');
      fireEvent.press(emailBox);
    });

    test('Email box should allow for input', () => {
      render(<ResetPasswordScreen/>);
      fireEvent.changeText(screen.getByPlaceholderText('Email'), EMAIL_TEXT)
    })
});


// Testing new password input boxe

const PASSWORD_TXT = 'password';

describe('Testing new password input', () => {
  it('should find the new password box via testId', () => {
  const testIdName = 'newPasswordBox';
  const {getByTestId} = render(<ResetPasswordScreen />);
  const foundButton = getByTestId(testIdName);
  expect(foundButton).toBeTruthy();
  });

    it('New password box is pressable', () => {
      const page = render(<ResetPasswordScreen/>);
      const newPasswordBox = page.getByTestId('newPasswordBox');
      fireEvent.press(newPasswordBox);
    });

    test('New password box should allow for input', () => {
      render(<ResetPasswordScreen/>);
      fireEvent.changeText(screen.getByPlaceholderText('New password'), PASSWORD_TXT)
    })
});


// Testing reset password button
describe('Testing reset password button', () => {
    it('should find the button via testId', () => {
    const testIdName = 'resetPasswordButton';
    const {getByTestId} = render(<ResetPasswordScreen />);
    const foundButton = getByTestId(testIdName);
    expect(foundButton).toBeTruthy();
    });

    it('button should say Reset Password', () => {
        const resetPasswordText = 'Reset Password'; // spaces around Login! because that's how it appears in code
        const notFoundText = 'error :(';
        const {toJSON, getByTestId, queryByText} = render(<ResetPasswordScreen />);
        const foundTextElement = getByTestId('resetPasswordButtonText');
        const notFoundTextElement = queryByText(notFoundText);
        expect(foundTextElement.props.children).toEqual(resetPasswordText);
        expect(notFoundTextElement).toBeNull();
        expect(toJSON()).toMatchSnapshot();
      });

      it('reset password button is pressable', () => {
        const navigation = {navigate: () =>{}}
        const page = render(<ResetPasswordScreen navigation={navigation}/>);
        const resetPasswordButton = page.getByTestId('resetPasswordButton');
        fireEvent.press(resetPasswordButton);
      })

      it('reset password navigates to login screen', () => {
            const navigation = {navigate: () =>{}}
            const page = render(<ResetPasswordScreen navigation={navigation}/>);
            jest.spyOn(navigation, 'navigate')
            const resetPasswordButton = page.getByTestId('resetPasswordButton');
            fireEvent.press(resetPasswordButton);
            expect(navigation.navigate).toHaveBeenCalledWith("Login")
          });
});

// Testing return to login button
describe('Testing return to login button', () => {
    it('should find the button via testId', () => {
    const testIdName = 'returnButton';
    const {getByTestId} = render(<ResetPasswordScreen />);
    const foundButton = getByTestId(testIdName);
    expect(foundButton).toBeTruthy();
    });

    it('button should say return to login', () => {
        const returnText = 'Return to login'; // spaces around Login! because that's how it appears in code
        const notFoundText = 'error :(';
        const {toJSON, getByTestId, queryByText} = render(<ResetPasswordScreen />);
        const foundTextElement = getByTestId('returnText');
        const notFoundTextElement = queryByText(notFoundText);
        expect(foundTextElement.props.children).toEqual(returnText);
        expect(notFoundTextElement).toBeNull();
        expect(toJSON()).toMatchSnapshot();
      });

      it('return to login button is pressable', () => {
        const navigation = {navigate: () =>{}}
        const page = render(<ResetPasswordScreen navigation={navigation}/>);
        const returnButton = page.getByTestId('returnButton');
        fireEvent.press(returnButton);
      })

      it('return to login button navigates to login screen', () => {
            const navigation = {navigate: () =>{}}
            const page = render(<ResetPasswordScreen navigation={navigation}/>);
            jest.spyOn(navigation, 'navigate')
            const returnButton = page.getByTestId('returnButton');
            fireEvent.press(returnButton);
            expect(navigation.navigate).toHaveBeenCalledWith("Login")
          });
});

// // Testing forgot password pressable

// describe('Testing forgot password', () => {
//   it('should find forgot password via testId', () => {
//   const testIdName = 'forgotPassword';
//   const {getByTestId} = render(<LoginScreen />);
//   const foundButton = getByTestId(testIdName);
//   expect(foundButton).toBeTruthy();
//   });

//   it('Forgot password is pressable', () => {
//     const navigation = {navigate: () =>{}}
//     const page = render(<LoginScreen navigation={navigation}/>);
//     const forgotPassword = page.getByTestId('forgotPassword');
//     fireEvent.press(forgotPassword);
//   });
  
//   it('Forgot password navigates to reset password screen', () => {
//     const navigation = {navigate: () =>{}}
//     const page = render(<LoginScreen navigation={navigation}/>);
//     jest.spyOn(navigation, 'navigate')
//     const forgotPassword = page.getByTestId('forgotPassword');
//     fireEvent.press(forgotPassword);
//     expect(navigation.navigate).toHaveBeenCalledWith("Reset Password")
//   });
// })


// // Testing create account pressable

// describe('Testing create account', () => {
//   it('should find create account via testId', () => {
//   const testIdName = 'createAccount';
//   const {getByTestId} = render(<LoginScreen />);
//   const foundButton = getByTestId(testIdName);
//   expect(foundButton).toBeTruthy();
//   });

//   it('Create account is pressable', () => {
//     const navigation = {navigate: () =>{}}
//     const page = render(<LoginScreen navigation={navigation}/>);
//     const createAccount = page.getByTestId('createAccount');
//     fireEvent.press(createAccount);
//   });
  
//   it('Create account navigates to create account screen', () => {
//     const navigation = {navigate: () =>{}}
//     const page = render(<LoginScreen navigation={navigation}/>);
//     jest.spyOn(navigation, 'navigate')
//     const createAccount = page.getByTestId('createAccount');
//     fireEvent.press(createAccount);
//     expect(navigation.navigate).toHaveBeenCalledWith("Create Account 1")
//   });
// })
