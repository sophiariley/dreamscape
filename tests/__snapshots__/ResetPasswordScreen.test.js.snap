// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly across screens 1`] = `
<View
  style={
    {
      "alignContent": "center",
      "backgroundColor": "white",
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <RCTScrollView
    style={
      {
        "flex": 1,
      }
    }
  >
    <View>
      <View
        style={
          {
            "height": 1334,
            "width": 750,
          }
        }
      >
        <Text
          style={
            {
              "alignSelf": "center",
              "color": "#F8C98A",
              "fontSize": 40,
              "top": 150,
            }
          }
        >
           Reset Password 
        </Text>
        <View
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
              "paddingTop": 200,
            }
          }
        >
          <Text
            style={
              {
                "alignSelf": "flex-start",
                "color": "#3A6496",
                "fontSize": 20,
                "left": 40,
              }
            }
          >
            Enter the email associated with your
          </Text>
          <Text
            style={
              {
                "alignSelf": "flex-start",
                "color": "#3A6496",
                "fontSize": 20,
                "left": 40,
              }
            }
          >
            account:
          </Text>
          <TextInput
            onChangeText={[Function]}
            placeholder="Email"
            style={
              {
                "backgroundColor": "#DADADA",
                "borderRadius": 5,
                "fontSize": 20,
                "height": "20%",
                "marginBottom": 10,
                "marginTop": 5,
                "opacity": 0.5,
                "paddingLeft": 10,
                "width": "80%",
              }
            }
            testID="emailBox"
            value=""
          />
          <Text
            style={
              {
                "alignSelf": "flex-start",
                "color": "#3A6496",
                "fontSize": 20,
                "left": 40,
              }
            }
          >
            Enter new password
          </Text>
          <TextInput
            onChangeText={[Function]}
            placeholder="New password"
            style={
              {
                "backgroundColor": "#DADADA",
                "borderRadius": 5,
                "fontSize": 20,
                "height": "20%",
                "marginBottom": 10,
                "marginTop": 5,
                "opacity": 0.5,
                "paddingLeft": 10,
                "width": "80%",
              }
            }
            testID="newPasswordBox"
            value=""
          />
        </View>
        <View
          style={
            {
              "alignSelf": "center",
              "flex": 1,
              "height": "20%",
              "justifyContent": "center",
              "top": -40,
              "width": "70%",
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#D28A8E",
                "borderRadius": 5,
                "opacity": 1,
                "padding": 15,
                "width": "100%",
              }
            }
            testID="resetPasswordButton"
          >
            <Text
              style={
                {
                  "color": "#F6F6F6",
                  "fontSize": 20,
                }
              }
            >
              Reset Password
            </Text>
          </View>
        </View>
        <View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
            testID="returnButton"
          >
            <Text
              style={
                {
                  "alignSelf": "center",
                  "bottom": 100,
                  "color": "#3A6496",
                  "fontSize": 20,
                }
              }
            >
              Return to login
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
